### PR TITLE
[PoC] Regenerate buildSettings after preGenerateCommands

### PR DESCRIPTION
**Pull request by wilzbach**
Fixes https://github.com/dlang/dub/issues/1474

This fixes the problem, but obviously we should be smarter 

TODO:
- [ ] add test case (or add the the examples to the CI)
- [x] check whether there actually was a preGenerateCommand run and only do the regeneration then
- [ ] try to find a better solution than this

FWIW it looks like even with the simple hello world example the buildSettings regeneration is already regenerated 4 times before.

### CODE REVIEW

File: examples/generated-sources/dub.json [hunk 1]
```diff
@@ -2,6 +2,6 @@
 	"name": "generated-sources",
 	"description": "Example of using pre generate commands to generate source code.",
 	"preGenerateCommands": [
-		"echo int fun() { return 42; } > source/test.d"
+		"echo 'int fun() { return 42; }' > source/test.d"
```
Comment (andre2007) [inline examples/generated-sources/dub.json]: Why are the ticks added? I just tested it and the file `test.d` is created as
```d
'int fun() { return 42; }' 
```

**Comment by andre2007** [comment]
@wilzbach 
I tested the PR and get a linker error the first time:

C:\Users\pc\git\dub\examples\generated-sources>dub build
Running pre-generate commands for generated-sources...
Performing "debug" build using dmd for x86_64.
generated-sources ~master: building configuration "application"...
Linking...
lld-link: error: undefined symbol: _D4test3funFZi
>>> referenced by C:\Users\pc\git\dub\examples\generated-sources\source\app.d:6
>>>               .dub\build\application-debug-windows-x86_64-dmd_2089-F8E1F3627CAEA609227D49EB76DB892B\generated-sources.obj:(_Dmain)
Error: linker exited with status 1
dmd failed with exit code 1.

it works only the second time.

File: examples/generated-sources/dub.json (hunk 1 above)
Comment (veelo) [reply examples/generated-sources/dub.json]: I can confirm that the ticks cause problems, at least on Windows.

**Comment by Geod24** [comment]
Fixed in #2035

**Pull request closed without merging**